pkg/liqo-controller-manager/networking/external-network/wireguard/utils.go

```diff
@@ -143,6 +143,27 @@ func ensureKeysSecret(ctx context.Context, cl client.Client, wgObj metav1.Object
 	}
 }
 
+func checkExistingKeysSecret(ctx context.Context, cl client.Client, secretName, namespace string) error {
+	var s corev1.Secret
+	if err := cl.Get(ctx, types.NamespacedName{Name: secretName, Namespace: namespace}, &s); err != nil {
+		return err
+	}
+
+	// check labels
+	if s.Labels == nil {
+		return fmt.Errorf("missing labels in secret %q", secretName)
```

Comment:
I would make the error more talkative indicating which labels are expected in the secret, so something like the following:

```suggestion
		return fmt.Errorf("mandatory labels %q: "true" and %q are missing in secret %q", consts.GatewayResourceLabel, consts.RemoteClusterID, secretName)
```